core/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridAggregator.java

```diff
@@ -66,6 +70,28 @@ public boolean needsScores() {
     }
 
     @Override
+    public void collectBucket(LeafBucketCollector subCollector, int doc, long bucketOrd) throws IOException {
+        bucketCentroids = bigArrays.grow(bucketCentroids, bucketOrd + 1);
+        super.collectBucket(subCollector, doc, bucketOrd);
+    }
+
+    protected final void adjustCentroid(long bucketOrd, long geohash) {
+        final int numDocs = getDocCounts().get(bucketOrd);
+        final GeoPoint oldCentroid = new GeoPoint();
+        final GeoPoint nextLoc = new GeoPoint();
+
+        if (numDocs > 1) {
+            final long curCentroid = bucketCentroids.get(bucketOrd);
+            oldCentroid.resetFromGeoHash(curCentroid);
```

Comment:
Thanks @colings86! This sets up for easy integration w/ geopointV2 coming in 2.1. I also ran benchmarks for this hash conversion (to/from) prior to a lucene commit which result in the following performance:  Avg computation: 19.740268909859154 ns  So its a good trade for the memory savings and compatibility w/ the next geopoint indexer.